src/index.ts

```diff
@@ -70,6 +77,15 @@ class Client {
             throw new Error('A port is required');
         }
 
+        this.isDebug = false;
+
+        this.debug = debuglog('dats', () => {
```

Comment:
This logic should be already handled by the debug log module. If the env var corresponding to the module you want to debug is present, then a call to `debug` will have an output, otherwise it's a no op. Is there some custom use case I am missing here? I think we should just use debug log and remove the possibility to override it. The debug should be used only or local troubleshooting.

Reply:
I want to allow the user to pass a custom log function